rust/src/webhooks.rs

```diff
@@ -86,10 +86,13 @@ impl Webhook {
             .filter_map(|x| x.split_once(','))
             .filter(|x| x.0 == SIGNATURE_VERSION)
             .any(|x| {
-                x.1.bytes()
-                    .zip(expected_signature.bytes())
-                    .fold(0, |acc, (a, b)| acc | (a ^ b))
-                    == 0
+                (x.1.len() == expected_signature.len())
+                    && (x
+                        .1
+                        .bytes()
+                        .zip(expected_signature.bytes())
+                        .fold(0, |acc, (a, b)| acc | (a ^ b))
+                        == 0)
```

Comment:
I think we can use [`Iterator::eq`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.eq) instead?

Reply:
No, it has to be constant time comparison, that's why it's done this way.